### PR TITLE
attempt to create animation for profile screen

### PR DESCRIPTION
**Pull request by spanastasia**

### CODE REVIEW

File: INTITA/Modules/Profile/ProfileViewController.swift [hunk 1]
```diff
@@ -56,6 +84,119 @@ class ProfileViewController: UIViewController, Storyboarded {
...
+                switch headerState {
+                case .decreased:
+                    opacity = 0
+                case .normal:
+                    opacity = 1
+                }
+                headerContentView.layer.sublayers?[3].opacity = opacity
+                headerContentView.layer.sublayers?[4].opacity = opacity
+                headerContentView.layer.sublayers?[5].opacity = opacity
+            }
```
Comment (oleksiikolomiietssnapp) [inline INTITA/Modules/Profile/ProfileViewController.swift:143]: animator.addCompletion { [weak self] _ in
guard let self ....
Please take a look.

File: INTITA/Modules/Profile/ProfileViewController.swift [hunk 2]
```diff
@@ -56,6 +84,119 @@ class ProfileViewController: UIViewController, Storyboarded {
...
+                        if tableView.contentOffset.y + 100 + currentTableViewContentYOffset < tableView.contentSize.height - tableView.bounds.size.height {
+                            tableView.setContentOffset(CGPoint(x: 0, y: tableView.contentOffset.y - currentTableViewContentYOffset + 100), animated: true)
+                        } else {
+                            tableView.scrollToRow(at: IndexPath(row: 4, section: 0), at: .bottom, animated: true)
+                        }
+                    }
+                case .decreased:
+                    tableView.scrollToRow(at: IndexPath(row: 0, section: 0), at: .top, animated: true)
+                }
+                headerState.toggle()
```
Comment (oleksiikolomiietssnapp) [inline INTITA/Modules/Profile/ProfileViewController.swift:122]: Please wrap it to the separate function. e.g. completeTheAnimation()

File: INTITA/Modules/Profile/ProfileViewController.swift [hunk 3]
```diff
@@ -56,6 +84,119 @@ class ProfileViewController: UIViewController, Storyboarded {
...
+            } else {
+                animator.isReversed = true
+                animator.startAnimation()
+                if headerState == .normal {
+                    tableView.setContentOffset(CGPoint(x: 0, y: currentTableViewContentYOffset), animated: true)
+                } else {
+                    tableView.setContentOffset(CGPoint(x: 0, y: 100), animated: true)
+                }
```
Comment (oleksiikolomiietssnapp) [inline INTITA/Modules/Profile/ProfileViewController.swift:130]: separate function. e.g. reverseAnimation()

**Comment by oleksiikolomiietssnapp** [comment]
Table view bottom constraint should be to superview.
Try to use CABasicAnimation for the layers.
```
                let fadeAnimation = CABasicAnimation()
                fadeAnimation.fromValue = 1.0
                fadeAnimation.toValue = 0.0
                headerContentView.layer.sublayers?[3].add(fadeAnimation, forKey: "opacity")
                headerContentView.layer.sublayers?[4].add(fadeAnimation, forKey: "opacity")
                headerContentView.layer.sublayers?[5].add(fadeAnimation, forKey: "opacity")
```

**Review by oleksiikolomiietssnapp — APPROVED** [review]

File: INTITA/Modules/Profile/ProfileViewController.swift [hunk 4]
```diff
@@ -125,7 +279,13 @@ extension ProfileViewController: ProfileViewModelDelegate {
 
 extension ProfileViewController: ProfileCoordinatorAlertPresenter {
     func showAlert() {
-        showAlert(message: "comming_soon".localized)
+        showAlert(header: "Oops!", message: "comming_soon".localized)
```
Comment (svitlana-korostelova) [inline INTITA/Modules/Profile/ProfileViewController.swift:282]: showAlert()  would be enough

File: INTITA/Modules/Profile/ProfileViewController.swift [hunk 5]
```diff
@@ -107,7 +261,7 @@ extension ProfileViewController: UITableViewDelegate {
     func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
         switch indexPath.row {
         case rowNumber - 1:
-            return 86
+            return 106
```
Comment (svitlana-korostelova) [inline INTITA/Modules/Profile/ProfileViewController.swift:264]: what about this magic number?

File: INTITA/Modules/Profile/ProfileViewController.swift [hunk 6]
```diff
@@ -25,23 +45,33 @@ class ProfileViewController: UIViewController, Storyboarded, AlertAcceptable {
...
+        initialHeaderHeight = headerContentView.frame.height
+
+        animator.isInterruptible = true
+        gestureRecognizer.addTarget(self, action: #selector(handleGesture))
+        gestureRecognizer.delegate = self
+        view.addGestureRecognizer(gestureRecognizer)
+        
+        headerView.addSubview(headerContentView)
+        headerContentView.translatesAutoresizingMaskIntoConstraints = false
+        headerContentView.shadowed(shadowOffset: CGSize(width: 0, height: 7))
```
Comment (svitlana-korostelova) [inline INTITA/Modules/Profile/ProfileViewController.swift:66]: probably better extract to func customization of the header view

**Pull request merged**